complete/src/main/java/com/example/restservice/dao/Queue.java

```diff
@@ -17,24 +26,21 @@
   @GenericGenerator(name = "uuid2", strategy = "uuid2")
   private String queueId;
 
+  private boolean isPasswordProtected;
+  private int queuePassword;
   @OneToMany(cascade = CascadeType.ALL, mappedBy = "queue")
   private List<User> users;
 
-  public List<User> getUsers() {
-    return users;
-  }
-
   protected Queue() {}
 
-  public String getQueueName() {
-    return queueName;
+  public Queue(String queueName)
+  {
+    this.queueName=queueName;
   }
 
-  public Queue(String queueName) {
+  public Queue(String queueName, boolean isPasswordProtected, int queuePassword) {
```

Comment:
Let's not take the queuePassword as an argument. Instead we can have a private function within the class that generates the random number, and call it from within the constructor.

Reply:
i thought of doing like that, but sonarlint advise to re-use Random object rather than creating new every time as the latter is inefficient. So I moved it to Business Service class to generate random pin using same random object instance. Thought of keep bean class clean without any non persisting variables.

Do you want to tie this back to bean class ? Then what do you recommend for handling random object ? Create a new instance in function every time or use something like a static member variable ?

Reply:
Keeping the Random object as `static final` inside the Queue class should solve that. 

The big question is, where should the logic lie? I was thinking that it's tied to the queue, so then the Queue class would have that logic as a private function, and constructor would call it. Later we will have a `public void refreshPin()` that would also call it.

But you are of the opinion that this is business logic, and it should be help in the service layer. Business logic would deal with things like when to call refresh queue, if the queue has to be password protected or not etc

@thehamzarocks I'll leave it to you to make the final judgement.

Reply:
I've no strong opinion, but keeping it within the Queue class means different (potential) service implementations wouldn't need to have their own PIN generation logic.

However, should we decide to change the PIN behaviour in the future (possibly allowing the user to enter his own PIN), then it might make more sense to keep the logic in the service class.

I'll leave it up to you, @avinashkris9 to decide.